src/tests/hash_functions/validation/Makefile.validation

```diff
@@ -0,0 +1,50 @@
+MOD_PATH = -I../../../
+
+FFLAGS = -O3
+CFLAGS = -O3
+CXXFLAGS = -O3
+LIBDIRS = -L./
+LIBS = -lc_hash
+INCLUDE_DIRS = -I./
+
+all: generate_hash_arrays generate_key_array hash_validity_test
+
+generate_key_array: generate_key_array.f90
+	$(FC) $(FFLAGS) generate_key_array.f90 -o generate_key_array
+
+hash_validity_test: hash_validity_test.f90
+	$(FC) $(FFLAGS) -L../../../ -lstdlib $(MOD_PATH) \
+	hash_validity_test.f90 -o hash_validity_test
+
+generate_hash_arrays: generate_hash_arrays.o ./libc_hash.a
+	$(CXX) $(CXXFLAGS) $(LIBDIRS) generate_hash_arrays.o \
```

Comment:
So I believe we have a c++ dependency? That will need some discussion. In principle I think it's a good idea, to help us support a greater range of functionality, as well as for situations like this.

Reply:
Realize that one of the C codes assumes thst the compiler must be either gcc or MSVC, and cannot be compiled as is with Intel's C compiler.

Reply:
Thanks for flagging that (I recall it is also mentioned in the documentation). I'll be interested to hear community opinions on the best approach here.

Reply:
It is not a problem for me if the C/C++ dependency is only for testing. It is actually probably better than providing a set of arrays to the hash functions and checking them againt hard-coded values.